workflow/scripts/additional_functionality.py

```diff
@@ -9,115 +9,79 @@
-                extendable_index = c.df.index[valid_components & c.df.p_nom_extendable]
+                existing_index = c.df.index[
+                    valid_components & ~c.df[attr + "_nom_extendable"]
+                ]
+                extendable_index = c.df.index[
+                    valid_components & c.df[attr + "_nom_extendable"]
+                ]
 
-                existing_capacity = c.df.loc[existing_index, "p_nom"].sum()
+                existing_capacity = c.df.loc[existing_index, attr + "_nom"].sum()
 
                 logger.info(
-                    f"Existing {c.name} {carrier} capacity in {ct}: {existing_capacity} MW"
+                    f"Existing {c.name} {carrier} capacity in {ct}: {existing_capacity} {units}"
                 )
-                logger.info(
-                    f"Adding constraint on {c.name} {carrier} capacity in {ct} to be smaller than {limit} MW"
-                )
-
-                p_nom = n.model[c.name + "-p_nom"].loc[extendable_index]
 
-                lhs = p_nom.sum()
-
-                cname = f"capacity_maximum-{ct}-{c.name}-{carrier.replace(' ','-')}"
-                if limit - existing_capacity <= 0:
-                    n.model.add_constraints(lhs <= 0, name=f"GlobalConstraint-{cname}")
-                    logger.warning(
-                        f"Existing capacity in {ct} for carrier {carrier} already exceeds the limit of {limit} MW. Limiting capacity expansion for this investment period to 0."
-                    )
-                else:
+                nom = n.model[c.name + "-" + attr + "_nom"].loc[extendable_index]
+
+                lhs = nom.sum()
+
+                cname = f"capacity_{sense}-{ct}-{c.name}-{carrier.replace(' ','-')}"
+
+                if sense == "maximum":
+                    if limit - existing_capacity <= 0:
+                        n.model.add_constraints(
+                            lhs <= 0, name=f"GlobalConstraint-{cname}"
+                        )
+                        logger.warning(
+                            f"Existing capacity in {ct} for carrier {carrier} already exceeds the limit of {limit} MW. Limiting capacity expansion for this investment period to 0."
+                        )
+                    else:
+                        n.model.add_constraints(
+                            lhs <= limit - existing_capacity,
+                            name=f"GlobalConstraint-{cname}",
+                        )
+                elif sense == "minimum":
                     n.model.add_constraints(
-                        lhs <= limit - existing_capacity,
+                        lhs >= limit - existing_capacity,
                         name=f"GlobalConstraint-{cname}",
                     )
+                else:
+                    logger.error("sense {sense} not recognised")
+                    sys.exit()
+
                 if cname not in n.global_constraints.index:
```

Comment:
i'm not entirely sure what this line does :thinking: 

@JulianGeis , i think you implemented it for the post discretization? Would it be possible to pull it into the main loop and replace `if cname not in: do` with `if cname in: return`?

Reply:
The "else" was me in case "sense" is not specified correctly (a paranoid check).

Reply:
There is nothing like a "paranoid check" 😬

Reply:
:heart: paranoid checks